src/amo/components/SearchResult/styles.scss

```diff
@@ -19,6 +19,7 @@ $icon-default-size: 32px;
 
 .SearchResult-link {
   text-decoration: none;
+  word-wrap: anywhere;
```

Comment:
MDN says:

> [`break-word` is the] same as the anywhere value, with normally unbreakable words allowed to be broken at arbitrary points if there are no otherwise acceptable break points in the line, but soft wrap opportunities introduced by the word break are NOT considered when calculating min-content intrinsic sizes.

I suppose the last part of the sentence explains why it only works with `anywhere` here.